contextualized/regression/trainers.py

```diff
@@ -3,42 +3,67 @@
 """
 import numpy as np
 import pytorch_lightning as pl
+from contextualized.regression.datamodules import RegressionDataModule
 
 
 class RegressionTrainer(pl.Trainer):
     """
     Trains the contextualized.regression lightning_modules
     """
 
-    def predict_params(self, model, dataloader):
+    def predict_params(self, model, dataclass, pred_dataloader_type="full"):
         """
+        :param model: Model to use for predicting
+        :param dataclass: Dataloader or datamodule class to predict on
+        :param pred_dataloader_type:If dataclass is a datamodule, choose what dataset the predict_dataloader will use: 'test', 'train', 'val', 'full'
+
         Returns context-specific regression models
         - beta (numpy.ndarray): (n, y_dim, x_dim)
         - mu (numpy.ndarray): (n, y_dim, [1 if normal regression, x_dim if univariate])
         """
-        preds = super().predict(model, dataloader)
-        return model._params_reshape(preds, dataloader)
 
-    def predict_y(self, model, dataloader):
+        if type(dataclass) in (pl.LightningDataModule, RegressionDataModule):
+            dataclass.setup("predict", pred_dataloader_type)
+
+        preds = super().predict(model, dataclass)
+        return model._params_reshape(preds, dataclass)
+
+    def predict_y(self, model, dataclass, pred_dataloader_type="full"):
```

Comment:
Let's make it clear that this kwarg is specific to the datamodules and also shorten the name if possible.